layersvt/device_simulation.cpp

```diff
@@ -73,16 +75,54 @@ const uint32_t kLayerPropertiesCount = (sizeof(kLayerProperties) / sizeof(kLayer
 const VkExtensionProperties *kExtensionProperties = nullptr;
 const uint32_t kExtensionPropertiesCount = 0;
 
+#if !defined(__ANDROID__)
 const char *kSchemaDevsim100 = "https://schema.khronos.org/vulkan/devsim_1_0_0.json#";
+#endif
 
 // Environment variables defined by this layer ///////////////////////////////////////////////////////////////////////////////////
 
+#if defined(__ANDROID__)
+const char *const kEnvarDevsimFilename = "debug.vulkan.devsim.filepath";        // path of the configuration file to load.
+const char *const kEnvarDevsimDebugEnable = "debug.vulkan.devsim.debugenable";  // a non-zero integer will enable debugging output.
+const char *const kEnvarDevsimExitOnError = "debug.vulkan.devsim.exitonerror";  // a non-zero integer will enable exit-on-error.
+#else
 const char *const kEnvarDevsimFilename = "VK_DEVSIM_FILENAME";          // name of the configuration file to load.
 const char *const kEnvarDevsimDebugEnable = "VK_DEVSIM_DEBUG_ENABLE";   // a non-zero integer will enable debugging output.
 const char *const kEnvarDevsimExitOnError = "VK_DEVSIM_EXIT_ON_ERROR";  // a non-zero integer will enable exit-on-error.
+#endif
 
 // Various small utility functions ///////////////////////////////////////////////////////////////////////////////////////////////
 
+#if defined(__ANDROID__)
+#include <android/log.h>
+const uint32_t MAX_BUFFER_SIZE = 255;
+
+const char *AndroidGetEnv(const char *key) {
```

Comment:
Yes, I have noticed the similar code in screenshot but not sure where should the common location be.